### PR TITLE
[6575] like habit notification

### PR DESCRIPTION
**Pull request by df7734**
This PR includes: 
Adding likes to habit entity + controller https://github.com/ita-social-projects/GreenCity/issues/7316
Adding like notifications (without changing notification style) https://github.com/ita-social-projects/GreenCity/issues/6575

### CODE REVIEW

File: core/src/test/java/greencity/controller/HabitControllerTest.java [hunk 1]
```diff
@@ -30,9 +32,7 @@
 import org.springframework.test.web.servlet.MockMvc;
 import static greencity.ModelUtils.getPrincipal;
 import static org.mockito.Mockito.verify;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
```
Comment (HelenSotnik) [inline core/src/test/java/greencity/controller/HabitControllerTest.java:34]: You should better add new import instead using asterisks in imports: 
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;

File: dao/src/main/java/greencity/entity/Habit.java [hunk 2]
```diff
@@ -2,18 +2,7 @@
...
-import jakarta.persistence.Column;
-import jakarta.persistence.Entity;
-import jakarta.persistence.FetchType;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.GenerationType;
-import jakarta.persistence.Id;
-import jakarta.persistence.JoinColumn;
-import jakarta.persistence.JoinTable;
-import jakarta.persistence.ManyToMany;
-import jakarta.persistence.OneToMany;
```
Comment (HelenSotnik) [inline dao/src/main/java/greencity/entity/Habit.java:15]: The same with these imports, better to get them back and add specific import, instead of using asterisk.

File: service/src/main/java/greencity/service/HabitServiceImpl.java [hunk 3]
```diff
@@ -16,7 +17,7 @@
 import greencity.entity.HabitTranslation;
 import greencity.entity.Tag;
 import greencity.entity.User;
-import greencity.enums.Role;
```
Comment (HelenSotnik) [inline service/src/main/java/greencity/service/HabitServiceImpl.java:19]: Update imports to avoid asterisk.

File: service/src/test/java/greencity/ModelUtils.java [hunk 4]
```diff
@@ -219,21 +219,8 @@
...
 import java.util.Set;
-import static greencity.enums.NotificationType.ECONEWS_COMMENT;
-import static greencity.enums.NotificationType.ECONEWS_COMMENT_LIKE;
-import static greencity.enums.NotificationType.ECONEWS_COMMENT_REPLY;
-import static greencity.enums.NotificationType.ECONEWS_CREATED;
-import static greencity.enums.NotificationType.ECONEWS_LIKE;
-import static greencity.enums.NotificationType.EVENT_CANCELED;
-import static greencity.enums.NotificationType.EVENT_COMMENT;
-import static greencity.enums.NotificationType.EVENT_COMMENT_LIKE;
-import static greencity.enums.NotificationType.EVENT_COMMENT_REPLY;
```
Comment (HelenSotnik) [inline service/src/test/java/greencity/ModelUtils.java]: Same thing with imports.

File: service/src/test/java/greencity/service/HabitServiceImplTest.java [hunk 5]
```diff
@@ -59,10 +61,9 @@
 import java.util.Set;
 import java.util.ArrayList;
 import java.util.stream.Stream;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertNotEquals;
```
Comment (HelenSotnik) [inline service/src/test/java/greencity/service/HabitServiceImplTest.java:65]: Update imports

Comment (df7734) [reply service/src/test/java/greencity/service/HabitServiceImplTest.java:65]: Imports fixed. Thanks for review!

File: service/src/test/java/greencity/service/HabitServiceImplTest.java [hunk 6]
```diff
@@ -1250,4 +1256,33 @@ void checkAccessOfOwnerToCustomHabitThrowsUserHasNoPermissionToAccessExceptionTe
         verify(userRepo).findByEmail(user.getEmail());
         verify(habitRepo, times(0)).save(any(Habit.class));
     }
+
+    @Test
+    void likeTest() {
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/HabitServiceImplTest.java:1285]: please add `verify` to your test

Comment (RomaMocherniuk) [reply service/src/test/java/greencity/service/HabitServiceImplTest.java:1285]: Please add the tests when
`habitRepo.findById(habitId)` and  `userRepo.findById(habit.getUserId())` throw exception

Comment (RomaMocherniuk) [reply service/src/test/java/greencity/service/HabitServiceImplTest.java:1285]: And please add a text that we do not send a notification if `habitAuthor == null`

File: service/src/main/java/greencity/service/EcoNewsServiceImpl.java [hunk 7]
```diff
@@ -477,7 +477,7 @@ private void enhanceWithNewData(EcoNews toUpdate, UpdateEcoNewsDto updateEcoNews
             .findTagsByNamesAndType(updateEcoNewsDto.getTags(), TagType.ECO_NEWS),
             new TypeToken<List<Tag>>() {
             }.getType()));
-        if (updateEcoNewsDto.getImage() != null) {
+        if (StringUtils.isNotBlank(updateEcoNewsDto.getImage())) {
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EcoNewsServiceImpl.java]: it is better not to include changes that are not related to your PR in it

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 8]
```diff
@@ -48,7 +48,7 @@ public class EventCommentServiceImpl implements EventCommentService {
     private ModelMapper modelMapper;
     private final EventRepo eventRepo;
     private final RatingCalculation ratingCalculation;
-    private AchievementCalculation achievementCalculation;
+    private final AchievementCalculation achievementCalculation;
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java]: same here

File: service/src/main/java/greencity/service/HabitServiceImpl.java [hunk 9]
```diff
@@ -524,6 +533,45 @@ public void deleteCustomHabit(Long customHabitId, String ownerEmail) {
         habitRepo.save(toDelete);
     }
 
+    /**
+     * Method to like or dislike {@link Habit} specified by id.
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/HabitServiceImpl.java]: why do you use `HabitVO` in the interface docs and `Habit` here?
maybe it is better to inherit docs from the interface?

File: service/src/main/java/greencity/service/UserNotificationServiceImpl.java [hunk 10]
```diff
@@ -255,7 +255,9 @@ public void removeActionUserFromNotification(UserVO targetUserVO, UserVO actionU
                 notificationRepo.delete(notification);
                 return;
             }
-            notification.getActionUsers().remove(modelMapper.map(actionUserVO, User.class));
+            User user = modelMapper.map(actionUserVO, User.class);
+            notification.getActionUsers()
+                .removeIf(u -> u.getId().equals(user.getId()));
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/UserNotificationServiceImpl.java:261]: please add a test for `removeIf`

File: service/src/main/resources/notification.properties [hunk 11]
```diff
@@ -33,3 +33,6 @@ FRIEND_REQUEST_ACCEPTED_TITLE=Your friend request was accepted
 FRIEND_REQUEST_ACCEPTED={user} accepted your friend request.
 FRIEND_REQUEST_RECEIVED_TITLE=You have received a friend request
 FRIEND_REQUEST_RECEIVED={user} sent you a friend request.
+
+HABIT_LIKE_TITLE=Your habit received a like
```
Comment (RomaMocherniuk) [inline service/src/main/resources/notification.properties]: Where do you use it?

Comment (df7734) [reply service/src/main/resources/notification.properties]: Issues fixed. Thank you for review!

**Review by ushh789 — APPROVED** [review]

**Comment by RomaMocherniuk** [comment]
Please increase test coverage

**Review by Maryna-511750 — APPROVED** [review]

**Review by KizerovDmitriy — APPROVED** [review]
